tests/test_orquestra_device.py

```diff
@@ -291,3 +420,393 @@ def circuit():
+
+            dev = qml.device(dev, wires=2)
+
+            with qml.tape.QuantumTape() as tape1:
+                qml.expval(qml.Identity(wires=[0]))
+                qml.expval(qml.PauliZ(wires=[1]))
+
+            with qml.tape.QuantumTape() as tape2:
+                qml.expval(qml.Identity(wires=[0]))
+
+            res = dev.batch_execute([tape1, tape2])
+
+            assert np.allclose(res[0], np.array([1, test_batch_res0]))
+            assert np.allclose(res[1], np.array([1]))
+
+    @pytest.mark.parametrize("dev", ["orquestra.forest", "orquestra.qiskit", "orquestra.qulacs"])
+    def test_identity_multiple_batched(self, dev):
+        """Test computing the expectation value of the identity for multiple
+        return values."""
+        dev = qml.device(dev, wires=2)
+
+        with qml.tape.QuantumTape() as tape1:
+            qml.expval(qml.Identity(wires=[0]))
+            qml.expval(qml.Identity(wires=[1]))
+
+        res = dev.batch_execute([tape1])
+        assert len(res) == 1
+        assert np.allclose(res[0], np.array([1, 1]))
+
+    @pytest.mark.parametrize("keep", [True, False])
+    def test_batch_exec(self, keep, tmpdir, monkeypatch, test_batch_result):
+        """Test that the batch_execute method returns the desired result and
+        that the result preserves the order in which circuits were
+        submitted."""
+        dev = qml.device("orquestra.forest", wires=3, keep_files=keep)
+
+        with qml.tape.QuantumTape() as tape1:
+            qml.expval(qml.PauliZ(wires=[0]))
+
+        with qml.tape.QuantumTape() as tape2:
+            qml.RX(0.432, wires=0)
+            qml.RY(0.543, wires=0)
+            qml.expval(qml.PauliZ(wires=[0]))
+
+        with qml.tape.QuantumTape() as tape3:
+            qml.RX(0.432, wires=0)
+            qml.expval(qml.PauliZ(wires=[0]))
+
+        circuits = [tape1, tape2, tape3]
+
+        test_uuid = "1234"
+        assert not os.path.exists(tmpdir.join(f"expval-{test_uuid}-0.yaml"))
+
+        with monkeypatch.context() as m:
+            m.setattr(pennylane_orquestra.cli_actions, "user_data_dir", lambda *args: tmpdir)
+
+            # Disable submitting to Orquestra by mocking Popen
+            m.setattr(subprocess, "Popen", lambda *args, **kwargs: MockPopen())
+            m.setattr(
+                pennylane_orquestra.orquestra_device,
+                "loop_until_finished",
+                lambda *args, **kwargs: test_batch_result,
+            )
+
+            # Disable random uuid generation
+            m.setattr(uuid, "uuid4", lambda *args: test_uuid)
+
+            res = dev.batch_execute(circuits)
+
+            # Correct order of results is expected
+            assert np.allclose(res[0], test_batch_res0)
+            assert np.allclose(res[1], test_batch_res1)
+            assert np.allclose(res[2], test_batch_res2)
+            file_kept = os.path.exists(tmpdir.join(f"expval-{test_uuid}-0.yaml"))
+
+            assert file_kept if keep else not file_kept
```

Comment:
I see you snuck in an extra test for keeping files here. :wink:

Reply:
Yep!